packages/core/admin/admin/src/content-manager/history/components/tests/VersionHeader.test.tsx

```diff
@@ -1,89 +1,203 @@
 import * as React from 'react';
 
-import { render as renderRTL, screen } from '@tests/utils';
+import { RenderOptions, render as renderRTL, screen } from '@tests/utils';
+import { Route, Routes } from 'react-router-dom';
 
 import { type HistoryContextValue, HistoryProvider } from '../../pages/History';
 import { VersionHeader } from '../VersionHeader';
 
 import type { UID } from '@strapi/types';
 
-// Mocks
-const layout = {
-  contentType: {
-    info: {
-      singularName: 'kitchensink',
-    },
-    settings: {
-      mainField: 'title',
-    },
-  },
-};
-const selectedVersion = {
-  id: '26',
-  contentType: 'api::kitchensink.kitchensink' as UID.ContentType,
-  relatedDocumentId: 'pcwmq3rlmp5w0be3cuplhnpr',
-  createdAt: '2022-01-01T00:00:00Z',
-  status: 'draft' as const,
-  schema: {},
-  locale: null,
-  data: {
-    title: 'Test Title',
-  },
-};
+const render = (
+  props: HistoryContextValue,
+  initialEntry: Exclude<RenderOptions['initialEntries'], undefined>[number]
+) => {
+  const path =
+    props.layout.contentType.kind === 'singleType'
+      ? '/:collectionType/:slug/history'
+      : '/:collectionType/:slug/:id/history';
 
-const render = (props: HistoryContextValue) =>
-  renderRTL(
+  return renderRTL(
```

Comment:
do you know about `renderOptions.wrapper` and when to use it? It's not entirely important here at the moment, but if you wanted to `re-render` you'd have to pass the whole tree you've constructed here as opposed to just the component you're actually changing, for example if you wanted to change `headerId` to test it can/can't find stuff etc.

Thought it was worth mentioning incase you didn't know 👍🏼

Reply:
Yes I know about wrapper, I don't think I need it here though, since the things that change from test to test here are actually the wrappers, the component itself remains the same: `<VersionHeader headerId="123" />`

But also, I noticed I don't see wrapper at all in the RenderOptions type. I suppose it's because I'm using `@tests/utils` which probably already uses wrapper. So I guess I'd have to import from RTL directly if I wanted to use wrapper?

Reply:
Yeah I did say it doesn't strictly apply here 😊 it does exist but because we also support things like user event setup stuff and initialentires, it's under the property "renderOptions" of the second argument to render.